### PR TITLE
[chore][processor/tailsampling] Add help information for resolving trace dropped errors

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>

A [recent issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/29024) was created when a user was getting a high value for the `otelcol_processor_tail_sampling_sampling_trace_dropped_too_early` metric. The solution was to increase the number of traces stored in memory to handle the load. This is done by using the `num_traces` configuration option. This was not clear from documentation or from the error metric's description.

I'm not sure of what location is the best place to communicate the solution to this error. The two options in my mind are to either information to the README, or add more information to the metric description itself. Any guidance here is appreciated to know what would be most clear to users.

### CODE REVIEW

File: processor/tailsamplingprocessor/metrics.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ var (
 	statCountTracesSampled       = stats.Int64("count_traces_sampled", "Count of traces that were sampled or not per sampling policy", stats.UnitDimensionless)
 	statCountGlobalTracesSampled = stats.Int64("global_count_traces_sampled", "Global count of traces that were sampled or not by at least one policy", stats.UnitDimensionless)
 
-	statDroppedTooEarlyCount    = stats.Int64("sampling_trace_dropped_too_early", "Count of traces that needed to be dropped the configured wait time", stats.UnitDimensionless)
+	statDroppedTooEarlyCount    = stats.Int64("sampling_trace_dropped_too_early", "Count of traces that needed to be dropped before the configured wait time", stats.UnitDimensionless)
```
Comment (crobert-1) [inline processor/tailsamplingprocessor/metrics.go:33]: Would it make sense to give more context in the description as to possible reasons why this error is hit? Or should we simply rely on the README to point users in the right direction?

**Review by jiekun — COMMENTED** [review]
Thanks for picking up this and perfecting the doc. Feel free to ignore my review if you feel it's not useful.

File: processor/tailsamplingprocessor/README.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ Multiple policies exist today and it is straight forward to add more. These incl
 
 The following configuration options can also be modified:
 - `decision_wait` (default = 30s): Wait time since the first span of a trace before making a sampling decision
-- `num_traces` (default = 50000): Number of traces kept in memory
+- `num_traces` (default = 50000): Number of traces kept in memory. Increasing this value can help in resolving `sampling_trace_dropped_too_early` errors.
```
[the remark this answers is not in the document]
Comment (crobert-1) [reply processor/tailsamplingprocessor/README.md]: Good idea. I've added an `FAQ` to the bottom of the README.

File: processor/tailsamplingprocessor/metrics.go (hunk 1 above)
Comment (crobert-1) [reply processor/tailsamplingprocessor/metrics.go:33]: I've added an FAQ section to the README, it should be enough at this point.

File: processor/tailsamplingprocessor/README.md [hunk 3]
```diff
@@ -427,3 +427,11 @@ As a rule of thumb, if you want to add probabilistic sampling and...
...
 [probabilistic_sampling_processor]: ../probabilisticsamplerprocessor
 [loadbalancing_exporter]: ../../exporter/loadbalancingexporter
+
+## FAQ
+
+**Q. Why am I seeing high values for the error metric `sampling_trace_dropped_too_early`?**
+
+**A.** This is likely a load issue. If the collector is processing more traces in-memory than the `num_traces` configuration
+option allows, some will have to be dropped before they can be sampled. Increasing the value of `num_traces` can
+help resolve this error.
```
Comment (jpkrohling) [inline processor/tailsamplingprocessor/README.md:437]:
```suggestion
help resolve this error, at the expense of increased memory usage.
```

Comment (jpkrohling) [reply processor/tailsamplingprocessor/README.md:437]: It's always a trade-off, it should be clear what are the consequences of increasing that value.

Comment (crobert-1) [reply processor/tailsamplingprocessor/README.md]: Agreed, thanks for calling this out. 👍

**Comment by jiekun** [comment]
The FAQ section looks good to me.

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**